#Perkenalan

Nama: Ansari Indrawan

Saya seorang data analis di The Look, salah satu platform e-commerce terbesar di planet Mars. Bertugas membuat laporan evaluasi aktivitas penjualan The Look.

#Objektif 
Mengevalusai aktivitas penjualan dan transaksi dari e-commerce The Look dengan menerapkan konsep statistik serta pemahaman bisnis

#Problem Statement

1. Bagaimana jumlah transaksi dengan status Complete tiap bulan sampai kuartal 3 di tahun 2022? berapa rata rata penjualan tiap kuartal.
2. Bagaimana total penjualan tiap bulan sampai kuartal 3 di tahun 2022? kalau mengalami kenaikan berapa rata rata penjualan tiap kuartal.
3. Berapa user yang melakukan transaksi dengan status Complete di tiap bulan dari kuartal 1 sampai kuartal 3 2022?
4. Apa kategori produk yang paling banyak dibeli di tiap bulannya sampai kuartal 3 tahun 2022?
5. Apa Kategori produk yang paling banyak mendapatkan pendapatan  di tiap bulannya sampai kuartal 3 tahun 2022?
6. Bagaimana keterkaitan antara jumlah user yang melakukan transaksi dengan membandingkan jumlah transaksi dan total penjualan? Pilih yang paling berkaitan

#Import pustaka

In [ ]:
from google.cloud import bigquery
from google.colab import auth

# Menghubungkan BigQuery dengan Colab


In [ ]:
auth.authenticate_user()
print('Authenticated')

project_id = "lat1-hacktive8" 
client = bigquery.Client(project=project_id)

Authenticated


# Menjawab problem 1
Transaksi The Look tahun 2022


In [ ]:
#menyimpan hasil data query pada variabel p1

p1 = client.query('''


#memilih kolom kolom dari tabel orders

SELECT 


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dari data kolom order_id
#menampilkan jumlah transaksi pada kolom transaction
#manmpilkan jumlahh item dalam setiap transaksi pada kolom total item sold

    COUNT(DISTINCT order_id) AS transaction,


#menunjukan sumber data dari tabel ordes

FROM `bigquery-public-data.thelook_ecommerce.orders`


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilka dataset dari variabel p1
p1

year  month  quarterly  transaction
0  2022      1          1          742
1  2022      2          1          720
2  2022      3          1          881
3  2022      4          2          856
4  2022      5          2          949
5  2022      6          2          924
6  2022      7          3         1032
7  2022      8          3         1140
8  2022      9          3         1147

##insight problem 1
Berdasarkan tabel diatas dapat dilihat data dari jumlah transaksi setiap bulan selama 3 kuartal pada tahun 2022. Terjadi penigkatan jumlah transaksi dari kuartal 1 sampai kuetal ketiga Transaksi terendah terjadi pada kuartal 1 yaitu dengan jumlah 720 transaksi, dan transaksi tertinggi terjadi pada kurtal ketiga yaitu sebesar 1147 trasaksi. Rata rata dari transaksi tiap kuartal adalah 781 pada kuartal 1, 909 pada kuartal 2, 1106 pada kuartal 3. 

# Menjawab Problem 2
Penjualaan The Look tahun 2022

In [ ]:
#menyimpan hasil data query pada variabel p2

p2 = client.query('''


#memilih kolom kolom dari tabel

SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#CONCAT digunakan untuk menggabungkan nilai total_sales dengan string'USD'
#ROUND digunakan untuk membulatkan hail pada kolom total_sales menjadi hanya 2 angka dibelakang koma
#SUM digunakan untuk menghitung total penjualan dari kolom sale price

    CONCAT(ROUND(SUM(sale_price), 2 ), ' ','USD') AS total_sales,


#memilih kolom dari tabel oerder items

FROM `bigquery-public-data.thelook_ecommerce.order_items`


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p2
p2

year  month  quarterly    total_sales
0  2022      1          1   67617.65 USD
1  2022      2          1   62862.36 USD
2  2022      3          1   74691.14 USD
3  2022      4          2   71157.74 USD
4  2022      5          2    80874.8 USD
5  2022      6          2    86017.1 USD
6  2022      7          3   89794.31 USD
7  2022      8          3    98879.5 USD
8  2022      9          3  101128.68 USD

## insight problem 2
Berdasarkan tabel diatas dapat dilihat data dari total penjualan setiap bulan sampai kuartal 3 pada tahun 2022. Terjadi penigkatan total penjualan dari kuartal 1 sampai kuetal ketiga, dengan penjualan terendah terjadi pada kuartal 1 bulan kedua yaitu sebesar 67617.65 USD, dan transaksi tertinggi terjadi pada kurtal ketiga bulan ke sembilan yaitu sebesar 101128.68 USD. Rata rata dari total penjualan The look tiap kuartalnya didapatkan sebesar 68,390.38 USD pada kuartal 1, 79,349.88 USD pada kuartal 2, 96,600.83 USD pada kuartal 3.

# Menjawab Problem 3
Analisis user The Look tahun 2022

In [ ]:
#menyimpan hasil data query pada variabel p3

p3 = client.query('''


SELECT

#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data perkuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dalam setiap group dari kolom user_id
#menampilkan jumlah transaksi pada kolom user_transaction

    COUNT(DISTINCT user_id) AS user_transaction,


#memilih kolom dari tabel oerder items

FROM `bigquery-public-data.thelook_ecommerce.order_items` 


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p3
p3

year  month  quarterly  user_transaction
0  2022      1          1               761
1  2022      2          1               732
2  2022      3          1               900
3  2022      4          2               865
4  2022      5          2               958
5  2022      6          2               936
6  2022      7          3              1044
7  2022      8          3              1153
8  2022      9          3              1158

## insight problem 3
Berdasarkan tabel diatas dapat dilihat data dari jumlah pengguna yang sukses bertransaksi di The Look e-commerce setiap bulan sampai kuartal 3 pada tahun 2022. Pengguna yang bertransaksi mengalami penigkatan dari kuartal pertama sampai kuartal ketiga, dengan transaksi terendah terjadi pada kuartal pertama bulan kedua yaitu sebesar 761 pengguna, sedangkan yang tertinggi terjadi pada kurtal ketiga bulan ke sembilan yaitu sebesar 1158 pengguna. Rata rata dari pengguna yang bertransaksi tiap kuartal adalah 797 pada kuartal 1, 919 pada kuartal 2, dan 1118 pengguna pada kuartal 3. 

# Menjawab Problem 4
Analisis produk The Look tahun 2022

In [ ]:
#menyimpan hasil data query pada variabel p4

p4 = client.query('''


#memilih kolom kolom dari hasil query 2
#dengan mengubah nama kolom kategory menjadi most_bought

SELECT 

    year,
    month,
    quarterly,
    category as most_bought,
    transaction_amount 


#menunjukan sumber data dari tabel hasil query 2

FROM (


#query 2
#memilih kolom kolom dari hasil query 1

      SELECT 
          category, 
          year, 
          month,
          quarterly,
          transaction_amount,


#fungsi ini digunakan untuk memberikan ranking pada setiap baris dalam hasil query 1  
#dengan diatur pada tahun 2022 dan setiap bulan pada kuartal 1-3 kemudian diurutkan berdasarkan tabel transaction amount dari yang terbesar
#hasil perankingan disimpan dalam kolom ranking 

          RANK() OVER (PARTITION BY year, month ORDER BY transaction_amount DESC) AS ranking


#menunjukan sumber data dari tabel hasil query 1

      FROM (


#query 1
#memilih kolom kolom dari tabel hasil gabungan antara kolom product_id dari order items dan kolom id dari product

            SELECT 


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

                p.category,
                EXTRACT(year FROM created_at) AS year,
                EXTRACT(month FROM created_at) AS month,
                EXTRACT(quarter FROM created_at) AS quarterly,
                COUNT(category) AS transaction_amount


#menunjukan sumber data dari tabel hasil JOIN antara order items dan product
#JOIN berfungsi untuk menggabungkan baris dari kolom spesifik yang memiliki nilai yang sama

            FROM `bigquery-public-data.thelook_ecommerce.order_items` AS oi
            JOIN `bigquery-public-data.thelook_ecommerce.products` AS p
            ON oi.product_id = p.id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

            WHERE status = 'Complete' 
                AND EXTRACT(year FROM created_at) = 2022 
                AND EXTRACT(quarter FROM created_at) BETWEEN 1 AND 3


#mengelompokan hasil query

            GROUP BY category, year, month, quarterly
          )


#mentapkan kriteria untuk data yang diquery, dimana baris yang ditampilkan hanya yg berperingkat 1
#mengurutkan baris berdasarkan bulan

      )WHERE ranking = 1
       ORDER BY month ASC;


''').to_dataframe()

In [ ]:
##menampilkan dataset dari variabel p4
p4

year  month  quarterly  most_bought  transaction_amount
0  2022      1          1    Intimates                  81
1  2022      2          1  Tops & Tees                  82
2  2022      3          1    Intimates                 119
3  2022      4          2        Jeans                  97
4  2022      5          2    Intimates                 111
5  2022      6          2        Jeans                 108
6  2022      7          3        Jeans                 110
7  2022      8          3    Intimates                 121
8  2022      9          3    Intimates                 138

##insight problem 4
Berdasarkan tabel diatas dapat dilihat data dari total produk yang dibeli berdasakrkan kategori di The Look e-commerce setiap bulan sampai kuartal 3 pada tahun 2022. Setelah dilakukan perankingan produk didapatkan peringkat dari category produk setiap bulannya, dimana ditmapilkan hanya pertingkat pertama dari tiap bulan berdasarkan tingkat penjualan tertinggi. Pada kuartal pertama produk paling banyak dibeli adalah kategory Intimates yaitu sebanyak 119 transaksi. Pada kuartal kedua produk yang paling banyak dibeli adalah kategori Intimates juga dengan jumlah 111 transaksi. Untuk kuartal ketiga produk yang paling banyak dibeli adalah kategori Intimates seklaigus menjadi kategori dengan transaksi tertinggi selama 3 kurtal yaitu 131 transaksi. 

# Menjawab Problem 5
Analisis penjualan The Look tahun 2022

In [ ]:
#menyimpan hasil data query pada variabel p4

p5 = client.query('''


#memilih kolom kolom dari hasil query 1
#dengan mengubah nama kolom category menjadi most_sold

SELECT 
    year,
    month,
    quarterly,
    category as most_sold,
    revenue,
    amount_sales 


#menunjukan sumber data dari tabel hasil query 2

FROM (


#query2
#memilih kolom kolom dari hasil query 1

      SELECT 
          category,
          year,
          month,
          quarterly,
          number_of_product,
          revenue,
          amount_sales,


#fungsi ini digunakan untuk memberikan ranking pada setiap baris dalam hasil query 1  
#dengan diatur pada tahun 2022 dan setiap bulan pada kuartal 1-3 kemudian diurutkan berdasarkan tabel total product dari yang terbesar
#hasil perankingan disimpan dalam kolom ranking 

      RANK() OVER (PARTITION BY year, month ORDER BY number_of_product DESC) AS ranking


#menunjukan sumber data dari tabel hasil query 1

      FROM (


#query 1
#memilih kolom kolom dari tabel hasil gabungan antara kolom product_id dari order items dan kolom id dari product

            SELECT 


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at
#menampilkan jumlah product pada kolom total_product
#ROUND digunakan untuk membulatkan hasil pada kolom amount_sales menjadi hanya 2 angka dibelakang koma
#SUM digunakan untuk menghitung total penjualan dari kolom retail price, cost, dan sale price
#menambahkan kolom revenue dari hasil pengurangan total tabel retail price dan total cost 

                p.category,
                EXTRACT(YEAR FROM created_at) AS year,
                EXTRACT(MONTH FROM created_at) AS month,
                EXTRACT(quarter FROM created_at) AS quarterly,
                COUNT(category) AS number_of_product,
                ROUND(SUM(retail_price)-SUM(cost), 2) AS revenue,
                ROUND(SUM(sale_price), 2) AS amount_sales


#menunjukan sumber data dari tabel hasil JOIN antara order items dan product
#JOIN berfungsi untuk menggabungkan tabel order_items pada kolom product id dengan tabel products dengan menggunakan kolom id.

            FROM `bigquery-public-data.thelook_ecommerce.order_items` AS oi
            JOIN `bigquery-public-data.thelook_ecommerce.products` AS p
            ON oi.product_id = p.id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

            WHERE status = 'Complete' AND EXTRACT(YEAR FROM created_at) = 2022 
                AND EXTRACT(QUARTER FROM created_at) BETWEEN 1 AND 3


#mengelompokan hasil query

            GROUP BY category, year, month, quarterly
            )


#mentapkan kriteria untuk data yang diquery, dimana baris yang ditampilkan hanya yg berperingkat 1
#mengurutkan baris berdasarkan bulan

      )WHERE ranking = 1
       ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p5
p5

year  month  quarterly    most_sold  revenue  amount_sales
0  2022      1          1    Intimates  1240.92       2674.49
1  2022      2          1  Tops & Tees  1722.82       3903.36
2  2022      3          1    Intimates  1771.83       3749.28
3  2022      4          2        Jeans  4618.07       9844.96
4  2022      5          2    Intimates  1719.88       3680.24
5  2022      6          2        Jeans  5120.20      10977.35
6  2022      7          3        Jeans  5111.15      10989.05
7  2022      8          3    Intimates  2000.58       4257.67
8  2022      9          3    Intimates  2438.83       5238.20

##insight problem 5
Berdasarkan tabel diatas dapat dilihat data dari total produk yang banyak memberikan keuntungan berdasakrkan kategori di The Look e-commerce setiap bulan sampai kuartal 3 pada tahun 2022. Setelah dilakukan perankingan produk didapatkan peringkat dari kategori produk setiap bulannya, dimana ditampilkan hanya pertingkat pertama dari tiap bulan berdasarkan tingkat hasil keuntungan tertinggi. Pada kuartal pertama produk paling banyak memberikan keuntungan adalah kategori Tops & Tees yaitu sebesar 1722.82 USD transaksi. Pada kuartal kedua produk yang paling banyak dibeli adalah kategori Jeans dengan jumlah 5120.20 USD. Untuk kuartal ketiga produk yang paling banyak dibeli adalah kategori Jeans yaitu dengan jumlah 5111.15 USD. Total omset The Look selama 3 kuartal berdasarkan kolom amount sales didapatkan sebesar 49,015.60 USD dan total keuntungan mencapai 27,144.28 USD . 

# Menjawab Problem 6
Analisis korelasi The look E-commerce 

In [ ]:
p6 = client.query('''

#memilih kolom kolom dari hasil query 1

SELECT 

#CORR digunakan untuk menhitung koefisien korelasi pertama antara kolom jumlah user dan jumlah transaksi, 
#kedua antara jumlah user dan total revenue

    CORR(number_of_user, transaction_amount) AS user_transaction_correlation,
    CORR(number_of_user, total_revenue) AS user_revenue_correlation


#menunjukan sumber data dari tabel hasil query 1

FROM (
      SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

          EXTRACT(year FROM o.created_at) AS year,
          EXTRACT(month FROM o.created_at) AS month,
          EXTRACT(quarter FROM o.created_at) AS quarterly, 


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dalam setiap group dari kolom user_id dan order id dari tabel order items
#menampilkan jumlah transaksi pada kolom user_transaction
#SUM digunakan untuk menghitung total pendapatan dari kolom sale price

          COUNT(DISTINCT oi.user_id) AS number_of_user,
          COUNT(DISTINCT oi.order_id) AS transaction_amount,
          SUM(sale_price) AS total_revenue


#menunjukan sumber data dari tabel hasil JOIN antara order items dan orders
#JOIN berfungsi untuk menggabungkan tabel order_items dengan tabel orders dengan menggunakan kolom order_id.

      FROM bigquery-public-data.thelook_ecommerce.order_items AS oi
      JOIN bigquery-public-data.thelook_ecommerce.orders AS o
      ON oi.order_id = o.order_id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

      WHERE o.status = 'Complete'
          AND EXTRACT(year FROM o.created_at) = 2022
          AND EXTRACT(quarter FROM o.created_at) BETWEEN 1 AND 3


#mengelompkan hasil query

        GROUP BY year, month, quarterly
        )

        
''').to_dataframe()

In [ ]:
p6

user_transaction_correlation  user_revenue_correlation
0                      0.999909                   0.97271

## insight problem 6
Berdasarkan hasil perhitungan korelasi dengan membandingkan korelasi antara jumlah pengguna dan jumlah transaksi serta korelasi antara jumlah pengguna dan total pendapatan di TheLook e-commerce dapat diamatin pola belanja dari pengguna. Pada kolom user transaction correlation didapatkan nilai korelasi sebesar 0.9999, ini menunjukan hubungan yang saling terkait antara jumlah pengguna dan jumlah transaksi. Dapat diartikan jika pengguna bertambah maka jumlah transaksi juga akan cenderung meningkat. Sama halnya dengan yang ditunjukan pada kolom user revenue correlation didapatkan nilai korelasi sebesar 0.9727 maka dapat diartikan jika pengguna bertambah maka total pendapatan juga akan cenderung meningkat. 

Dengan memilih fokus pada hasil korelasi anatara jumlah pengguna dan jumlah transaksi yang memiliki korelasi yang tinggi ini menunjukan pengguna dari TheLook e-commerce akan melakukan transaksi lebih banyak saat jumlah pengguna semakin banyak. Maka target dari TheLook adalah meningkatkan jumlah pengguna sehingga dapat meningkatkan transaksi e-commerce TheLook secara signifikan.

# Menjawab Poin Pertanyaan

1. Problem satemet yang saya definisikan dapat tercapai semua dan dapat diukur dari segi nilai yang ditampilkan pada tabel serta insight yang berikan pada setiap tabel hasil anailis yang disajikan dalam menjawab setiap problem. 

2. Berdasarkan hasil anlisis yang dilakukan maka untuk mencapai 250,000 USD pada awal kuartal 4 sangat sulit berdasarkan jumlah transaksi dan jumlah pengguna selama 3 kuartal. Dengan berdarakan penjualan kategori produk didapatkan hasil total pendapatan untuk produk yang paling banyak dibeli sebesar 27,144.28 USD dalam 3 kuartal, serta melihat juga berdasarkan penjualan didapatakn rata rata omset tiap kuartal selama 3 kuartal yaitu sebesar 68,890 USD, 79,849 USD, 96,600 USD. Maka akan tidak masuk akal jika ingin menargetkan pendapatan 250,000 USD pada awal kuartal 4 sedangkan omset selama kuartal 3 saja hanya 96,600 USD.

3. Berdasarkan hasil analisis yang dilakukan dengan 6 problem statement yang dibuat maka didapatkan:
- trend transaksi dari TheLook mengalami peningkatan setiap kurtalnya dengan melihat nilai rata - rata transaksi tiap kuartal selama 3 kuartal yaitu, 781 transaksi, 909 transaksi, 1106 transaksi. 
- Jumlah pengguna yang bertaransaksi pada TheLook juga mengalami peningkatan yang signifikan pada tiap kuartal selama 3 kuartal, Dengan melihat nilai rata rata tiap kuartal dengan jumlah transaksi adalah 797 pengguna, 919 pengguna, dan 1118 pengguna.
- Gross marchandise Value pada 3 kuartal didapatkan berdasarkan hasil penjualan selama 3 kuartal yang dirata rata tiap kuartal yaitu sebesar 68,890 USD, 79,849 USD, 96,600 USD. Dari nilai tersebut dapat dilihat trend dari penjualan selama 3 kuartal mengalami kenaikan.                                                
Berdasarkan ketiga kriteria yang sudah dijelaskan, maka perusahaan TheLook layak untuk mendapatkan investor baru karena selama 3 kuartal bisnis TheLook mengalami trend psoitif